openprocurement/auctions/dgf/models.py

```diff
@@ -40,17 +40,40 @@ def read_json(name):
 ORA_CODES = ORA_CODES[:]
 ORA_CODES[0:0] = ["UA-IPN", "UA-FIN"]
 
+CAV_CODES = read_json('cav.json')
+CPVS_CODES = read_json('cpvs.json')
+
 DGF_ID_REQUIRED_FROM = datetime(2017, 1, 1, tzinfo=TZ)
 
 
+class CPVCAVClassification(Classification):
+    scheme = StringType(required=True, default=u'CPV', choices=[u'CPV', u'CAV'])
+    id = StringType(required=True)
+
+    def validate_id(self, data, code):
+        if data.get('scheme') == u'CPV' and code not in CPV_CODES:
+            raise ValidationError(BaseType.MESSAGES['choices'].format(unicode(CPV_CODES)))
+        elif data.get('scheme') == u'CAV' and code not in CAV_CODES:
+            raise ValidationError(BaseType.MESSAGES['choices'].format(unicode(CAV_CODES)))
+        if code.find("00000-") > 0:
+            raise ValidationError('At least CPV/CAV classification class should be specified more precisely')
```

Comment:
'At least {} classification class (XXXX0000-Y) should be specified more precisely'.format(data.get('scheme'))